### ISSUE TITLE
query-orders sexp not working when orders translated in strings.tbl

### ISSUE THREAD

**Issue by Achille44** [scp-fs2open/fs2open.github.com#4076]
In missions, when a query-orders sexp uses as a parameter an order that has been translated in strings.tbl (id from 299 to 311), the following error shows in fast debug mode:

```
Assert: "order != -1"
File: hudsquadmsg.cpp
Line: 2425

ntdll.dll! NtWaitForSingleObject + 20 bytes
KERNELBASE.dll! WaitForSingleObjectEx + 142 bytes
fs2_open_21_4_0_x64_AVX-FASTDBG.exe! <no symbol>
fs2_open_21_4_0_x64_AVX-FASTDBG.exe! <no symbol>
fs2_open_21_4_0_x64_AVX-FASTDBG.exe! <no symbol>
fs2_open_21_4_0_x64_AVX-FASTDBG.exe! <no symbol>
fs2_open_21_4_0_x64_AVX-FASTDBG.exe! <no symbol>
fs2_open_21_4_0_x64_AVX-FASTDBG.exe! <no symbol>
fs2_open_21_4_0_x64_AVX-FASTDBG.exe! <no symbol>
fs2_open_21_4_0_x64_AVX-FASTDBG.exe! <no symbol>
fs2_open_21_4_0_x64_AVX-FASTDBG.exe! <no symbol>
fs2_open_21_4_0_x64_AVX-FASTDBG.exe! <no symbol>
fs2_open_21_4_0_x64_AVX-FASTDBG.exe! <no symbol>
fs2_open_21_4_0_x64_AVX-FASTDBG.exe! <no symbol>
fs2_open_21_4_0_x64_AVX-FASTDBG.exe! <no symbol>
fs2_open_21_4_0_x64_AVX-FASTDBG.exe! <no symbol>
fs2_open_21_4_0_x64_AVX-FASTDBG.exe! <no symbol>
fs2_open_21_4_0_x64_AVX-FASTDBG.exe! <no symbol>
fs2_open_21_4_0_x64_AVX-FASTDBG.exe! <no symbol>
fs2_open_21_4_0_x64_AVX-FASTDBG.exe! <no symbol>
fs2_open_21_4_0_x64_AVX-FASTDBG.exe! <no symbol>
fs2_open_21_4_0_x64_AVX-FASTDBG.exe! <no symbol>
fs2_open_21_4_0_x64_AVX-FASTDBG.exe! <no symbol>
fs2_open_21_4_0_x64_AVX-FASTDBG.exe! <no symbol>
fs2_open_21_4_0_x64_AVX-FASTDBG.exe! <no symbol>
KERNEL32.DLL! BaseThreadInitThunk + 20 bytes
[...]
[ This info is in the clipboard so you can paste it somewhere now ]


Use Debug to break into Debugger, Exit will close the application.

ntdll.dll! NtWaitForSingleObject + 20 bytes
KERNELBASE.dll! WaitForSingleObjectEx + 142 bytes
fs2_open_21_4_0_x64_AVX-FASTDBG.exe! <no symbol>
fs2_open_21_4_0_x64_AVX-FASTDBG.exe! <no symbol>
fs2_open_21_4_0_x64_AVX-FASTDBG.exe! <no symbol>
fs2_open_21_4_0_x64_AVX-FASTDBG.exe! <no symbol>
fs2_open_21_4_0_x64_AVX-FASTDBG.exe! <no symbol>
fs2_open_21_4_0_x64_AVX-FASTDBG.exe! <no symbol>
fs2_open_21_4_0_x64_AVX-FASTDBG.exe! <no symbol>
fs2_open_21_4_0_x64_AVX-FASTDBG.exe! <no symbol>
fs2_open_21_4_0_x64_AVX-FASTDBG.exe! <no symbol>
fs2_open_21_4_0_x64_AVX-FASTDBG.exe! <no symbol>
fs2_open_21_4_0_x64_AVX-FASTDBG.exe! <no symbol>
fs2_open_21_4_0_x64_AVX-FASTDBG.exe! <no symbol>
fs2_open_21_4_0_x64_AVX-FASTDBG.exe! <no symbol>
fs2_open_21_4_0_x64_AVX-FASTDBG.exe! <no symbol>
fs2_open_21_4_0_x64_AVX-FASTDBG.exe! <no symbol>
fs2_open_21_4_0_x64_AVX-FASTDBG.exe! <no symbol>
fs2_open_21_4_0_x64_AVX-FASTDBG.exe! <no symbol>
fs2_open_21_4_0_x64_AVX-FASTDBG.exe! <no symbol>
fs2_open_21_4_0_x64_AVX-FASTDBG.exe! <no symbol>
fs2_open_21_4_0_x64_AVX-FASTDBG.exe! <no symbol>
fs2_open_21_4_0_x64_AVX-FASTDBG.exe! <no symbol>
fs2_open_21_4_0_x64_AVX-FASTDBG.exe! <no symbol>
fs2_open_21_4_0_x64_AVX-FASTDBG.exe! <no symbol>
fs2_open_21_4_0_x64_AVX-FASTDBG.exe! <no symbol>
KERNEL32.DLL! BaseThreadInitThunk + 20 bytes
ntdll.dll! RtlUserThreadStart + 33 bytes

```

In some cases it can prevent the player to complete the mission (e.g. tutorial mission 1 of Diaspora).

**Comment by BMagnu** [comment]
This should have been fixed by #3918.
Given that this log is from 21.4, could you retest this with RC3 or a current nightly?

**Comment by Achille44** [comment]
Indeed, the problem is fixed in RC3. Thanks for the help!
